### PR TITLE
Display IPsec field in hexadecimal

### LINKED ISSUE #533: ICMPv6 packet could not be parsed with IPSec AH protocol configured
I am attempting to parse a pcap file by using Scapy as lib, and now I encounter a problem. The ICMPv6 packet, which has been sent via IPSec AH protocol, could not be shown with show()/show2().
Pcap file: 
[AH_IPV6.zip](https://github.com/secdev/scapy/files/805587/AH_IPV6.zip)
My code is easy like this:
    
    from scapy.utils import rdpcap
    from scapy.all import conf

    def main():
        packet_list = rdpcap("/media/sf_ubshare/pcap/AH_IPV6.PCAP")
        pkt2 = packet_list.__getitem__(9)
        pkt2.show2()
    

ICMPv6 packet is missing in show2()
    
###[ IPv6 ]### 
     version   = 6L
     tc        = 0L
     fl        = 0L
     plen      = 128
     nh        = AH Header
     hlim      = 240
     src       = 1::2:1
     dst       = 1::3:2
###[ AH ]### 
        nh        = ipv6
        payloadlen= 4
        reserved  = 0
        spi       = 0x9592d65b
        seq       = 1
        icv       = '\x03\xa77\xc1\xea<\xcf\xb9bO~m`\x00\x00\x00\x00@:@\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x01\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x00\x02\x80\x00I\xe2\x00d\x00\x00X\xa6\xa7\x10\x00\x0eJm\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./01234567'
        padding   = None
    
but in wireshark the ICMPv6 packet could be parsed successfully
image

It seems like Scapy considers the ICMPv6 packet as a icv field of last packet.
after some dug I suspect there might be a problem in /scapy/utils.py
    
    def read_packet(self, size=MTU):
        """return a single packet read from the file
        
        returns None when no more packets are available
        """
        hdr = self.f.read(16)
        if len(hdr) < 16:
            return None
        sec,usec,caplen,wirelen = struct.unpack(self.endian+"IIII", hdr)  #<-- this line
        s = self.f.read(caplen)[:size]
        return s,(sec,usec,wirelen) # caplen = len(s)

Test bed info:

- Scapy version: 2.3.3

- OS: Ubuntu 14.04.3 LTS

### PR DESCRIPTION
**Pull request by ThomasFaivre**
_ESPPlain still use StrField but it is an internal class so display is not really important.

Fixes #533

### CODE REVIEW

File: scapy/layers/ipsec.py [hunk 1]
```diff
@@ -47,8 +47,17 @@
...
+    ByteEnumField,
+    ByteField,
+    IntField,
+    PacketField,
+    ShortField,
+    StrField,
+    XIntField,
+    XStrField,
+    XStrLenField,
+)
```
Comment (gpotter2) [inline scapy/layers/ipsec.py]: This is quite ugly isn't it ? I guess removing the parenthesis + using only 2 lines would be easier to read

Comment (ThomasFaivre) [inline scapy/layers/ipsec.py]: It is a question of point of view ^^
I find this clearer, easier to sort and make diff shorter afterwards.
I can change though, it is quite easy!

Comment (gpotter2) [inline scapy/layers/ipsec.py]: Of course, I do prefer small imports ^^
What do others think ?

File: scapy/layers/ipsec.py [hunk 2]
```diff
@@ -47,8 +47,17 @@
 from scapy.config import conf, crypto_validator
 from scapy.data import IP_PROTOS
 from scapy.error import log_loading
-from scapy.fields import (ByteEnumField, ByteField, StrField, StrLenField,
-                          XIntField, IntField, ShortField, PacketField)
+from scapy.fields import (
```
Comment (p-l-) [inline scapy/layers/ipsec.py]: I would prefer without the parenthesis and with `\` at end of the lines:

```Python
from scapy.fields import ByteEnumField, ByteField, IntField, PacketField, \
    ShortField, StrField, XIntField, XStrField, XStrLenField
```

**Comment by ThomasFaivre** [comment]
Changed imports as requested.
Also changed imports from inet6 for consistency

**Pull request merged**